src/filters/RouteAccessControl.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @link http://www.diemeisterei.de/
+ * @copyright Copyright (c) 2023 diemeisterei GmbH, Stuttgart
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace dmstr\web\filters;
+
+use yii\filters\AccessControl;
+use yii\helpers\ArrayHelper;
+use yii\web\Controller;
+
+/**
+ * This is an AccessControl filter with a "hardcoded" rule, that checks access to the current route
+ * - Access can be granted to each "level" of the current route
+ * - the permission name(s) that will be checked, are the "levels" of the route with concatinated with self::routePartSeparator
+ * Example: 
+ * - route: 'user/admin/role/index'
+ * - self::routePartSeparator = '_'
+ * - checks:
+ *   - 'user'
+ *   - 'user_admin'
+ *   - 'user_admin_role'
+ *   - 'user_admin_role_index'
+ */
+class RouteAccessControl extends AccessControl
+{
+
+    public $routePartSeparator = '_';
+    public $routeCheckParams = [];
+
+    public function beforeAction($action)
+    {
+        $this->rules[] = $this->getRouteAcessControlRule();
```

Comment:
Let‘s add a index here e.g. routeAccess. Even if we don‘t need it (yet), it is better to have it.

Reply:
done in [9e8213d](https://github.com/dmstr/yii2-web/pull/7/commits/9e8213db34780982495749c09222ecd7433b60e1)